src/protocol_definitions.rs

```diff
@@ -181,32 +183,48 @@ impl From<&GetFwVersionResponse> for [u8; 60] {
         // Serialize component_info
         let mut offset = 4;
         for i in 0..response.header.component_count as usize {
-            let component = &response.component_info[i];
-            bytes[offset] = component.packed_byte;
-            bytes[offset + 1] = component.component_id;
-            bytes[offset + 2..offset + 4].copy_from_slice(&component.vendor_specific1.to_le_bytes());
-            bytes[offset + 4] = component.fw_version.major;
-            bytes[offset + 5..offset + 7].copy_from_slice(&component.fw_version.minor.to_le_bytes());
-            bytes[offset + 7] = component.fw_version.variant;
+            let component = &response
+                .component_info
+                .get(i)
+                .ok_or(ConversionError::ByteConversionError)?;
+            *bytes.get_mut(offset).ok_or(ConversionError::ByteConversionError)? = component.packed_byte;
+            *bytes.get_mut(offset + 1).ok_or(ConversionError::ByteConversionError)? = component.component_id;
+            bytes
+                .get_mut(offset + 2..offset + 4)
+                .ok_or(ConversionError::ByteConversionError)?
+                .copy_from_slice(&component.vendor_specific1.to_le_bytes());
+            *bytes.get_mut(offset + 4).ok_or(ConversionError::ByteConversionError)? = component.fw_version.major;
+            bytes
+                .get_mut(offset + 5..offset + 7)
+                .ok_or(ConversionError::ByteConversionError)?
+                .copy_from_slice(&component.fw_version.minor.to_le_bytes());
```

Comment:
this might benefit from a helper - `copy_from_mut_safe` - where you pass it the target and destination, and the range to copy. lots of duplication rn with the ByteConversionError stuff.